### PR TITLE
Update use-defused-xml.yaml

### PR DESCRIPTION
**Pull request by dlukeomalley**
Add OWASP reference and lead with the official recommendation of the Python documentation. Also, prefer back tic over ' for GitHub comment formatting.

### CODE REVIEW

**Review by chmccreery — APPROVED** [review]

File: c/lang/security/insecure-use-string-copy-fn.yaml [hunk 1]
```diff
@@ -1,19 +1,17 @@
 rules:
   - id: insecure-use-string-copy-fn
-    pattern-either:
-      - pattern: strcpy(...)
-      - pattern: strncpy(...)
+    patterns:
+      - pattern: $X(...)
+      - metavariable-regex:
```
Comment (dlukeomalley) [inline c/lang/security/insecure-use-string-copy-fn.yaml]: I suspect this is slow compared to the prior implementation.

Comment (inkz) [reply c/lang/security/insecure-use-string-copy-fn.yaml]: yep, gonna be slower, the rest is lgtm! :+1:

Comment (dlukeomalley) [reply c/lang/security/insecure-use-string-copy-fn.yaml]: @inkz If we take this approach, is this a simple autofix for us to add?

Comment (dlukeomalley) [reply c/lang/security/insecure-use-string-copy-fn.yaml]: @aryx Got any ideas for how I could approach this for both autofix + using metavariables in the rule message + being performant?

Comment (inkz) [reply c/lang/security/insecure-use-string-copy-fn.yaml]: @dlukeomalley it is easy to change func name https://semgrep.dev/s/XZ1A
but I am not 100% sure that arguments will be correct.
Also there is nice comment/pr from one of our users on this rule: https://github.com/returntocorp/semgrep-rules/pull/1824

**Review by kurt-r2c — APPROVED** [review]
messaging improvements 🙌

File: c/lang/security/insecure-use-string-copy-fn.yaml (hunk 1 above)
Comment (dlukeomalley) [reply c/lang/security/insecure-use-string-copy-fn.yaml]: I removed this change for now.

**Review by inkz — APPROVED** [review]

**Pull request merged**